### DISCUSSION TITLE
Missing keyword `AUTOINCREMENT` when `autoincrement=True` on sqlite3

### DISCUSSION THREAD

**Discussion by loynoir** [sqlalchemy/sqlalchemy#7636] [Usage Questions] [answered]
### Describe the bug

Using below on sqlite
```python
class Message(Base):
    __tablename__ = "messages"

    id = Column(Integer, primary_key=True, autoincrement=True)
    message = Column(String)
```

Expect table
```sql
CREATE TABLE "messages" (
   "id"	INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
   "message"	VARCHAR
);
```

But got below, keyword `AUTOINCREMENT` is missing
```sql
CREATE TABLE "messages" (
	"id"	INTEGER NOT NULL,
	"message"	VARCHAR,
	PRIMARY KEY("id")
);
```

### To Reproduce
1. On a **empty** sqlite3 database, **without** `sqlite_sequence` table
2. Insert first. Insert second. **Delete second**. Insert third. Expect third's id to be `3`, but got `2` 

<details>
  <summary></summary>
  <p>
 
```python
from sqlalchemy import create_engine
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker
from sqlalchemy import Column, Integer, String

engine = create_engine("sqlite:///./reproduce.db")
Base = declarative_base()
Session = sessionmaker(bind=engine)
session = Session()


class Message(Base):
    __tablename__ = "messages"

    id = Column(Integer, primary_key=True, autoincrement=True)
    message = Column(String)


Base.metadata.create_all(engine)
message1 = Message(message="Hello World! 1")
message2 = Message(message="Hello World! 2")
message3 = Message(message="Hello World! 3")

# simulate
session.add(message1)
session.commit()

session.add(message2)
session.commit()
session.delete(message2)
session.commit()

session.add(message3)
session.commit()

query = session.query(Message)
results = query.all()

assert results[1].message == 'Hello World! 3'

# Error: id is 2
assert results[1].id == 3
```
  </p>
</details>

 3. `rm reproduce.db`
 4. run `sqlite3 reproduce.db` with 
 ```sql
CREATE TABLE "messages" (
	"id"	INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
	"message"	VARCHAR
);
```
5. Run python code again. Now third's id is `3`

### Error
1. Surface error

Insert first. Insert second. **Delete second**. Insert third. Expect third's id to be `3`, but got `2` 

2. Root error

Open sqlitebrowser, can see generated table structure have no `AUTOINCREMENT` keyword.

```sql
CREATE TABLE "messages" (
	"id"	INTEGER NOT NULL,
	"message"	VARCHAR,
	PRIMARY KEY("id")
);
```

### Versions

- OS: archlinux
- Python: 3.9-3.10
- SQLAlchemy: 1.4.31
- Database: sqlite3
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): sqlite3

**Comment by CaselIT** [comment, marked as answer]
Hi.

That's the expected behaviour, since sqlite does not require it for an autoincrement primary key. You can force the use of it gia a flag. See the documentation https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#sqlite-auto-incrementing-behavior

**Comment by loynoir** [comment]
Wow, thanks! @CaselIT 

Not sure what does below mean.
```python
Table('sometable', metadata,
        Column('id', Integer, primary_key=True),
        sqlite_autoincrement=True)
```

But below works!
```python
class Message(Base):
    __tablename__ = "messages"
    __table_args__ = {'sqlite_autoincrement': True}
    
    id = Column(Integer, primary_key=True, autoincrement=True)
    message = Column(String)
```

Thanks!!!
